versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5771 @@
+* The special value `null` is allowed for undefined attributes.
+* Expressions are not allowed.
+
+A meta object is similar to a regular WDL object, except:
+* The `object` keyword is not required.
+* Its values must conform to the same metadata rules defined above.
+
+Note that, unlike the WDL `Object` type, metadata objects are not deprecated and will continue to be supported in future versions.
+
+#### Task Metadata Section
+
+```txt
+$meta = 'meta' $ws* '{' ($ws* $meta_kv $ws*)* '}'
+```
+
+This section contains task-level metadata. For example: author and contact email.
+
+#### Parameter Metadata Section
+
+```txt
+$parameter_meta = 'parameter_meta' $ws* '{' ($ws* $meta_kv $ws*)* '}'
+```
+
+This section contains metadata specific to input and output parameters. Any key in this section *must* correspond to a task input or output.
+
+```wdl
+task wc {
+  input {
+    File infile
+    Boolean lines_only = false
+    String? region
+  }
+
+  parameter_meta {
+    infile: {
+      help: "Count the number of lines in this file"
+    }
+    lines_only: { 
+      help: "Count only lines"
+    }
+    region: {
+      help: "Cloud region",
+      suggestions: ["us-west", "us-east", "asia-pacific", "europe-central"]
+    }
+  }
+
+  command {
+    wc ${true="-l", false=' ' lines_only} ${infile}
+  }
+
+  output {
+     String retval = stdout()
+  }
+
+  runtime {
+    container: "my_image:latest"
+  }
+}
+```
+
+### Examples
+
+#### Example 1: Simplest Task
+
+```wdl
+task hello_world {
+  command {echo "hello world"}
+}
+```
+
+#### Example 2: Inputs/Outputs
+
+```wdl
+task one_and_one {
+  input {
+    String pattern
+    File infile
+  }
+  command {
+    grep ${pattern} ${infile}
+  }
+  output {
+    File filtered = stdout()
+  }
+  runtime {
+    container: "my_image:latest"
+  }
+}
+```
+
+#### Example 3: Runtime/Metadata
+
+```wdl
+task runtime_meta {
+  input {
+    Int memory_mb
+    String sample_id
+    String param
+  }
+  command {
+    java -Xmx${memory_mb}M -jar task.jar -id ${sample_id} -param ${param} -out ${sample_id}.out
+  }
+  output {
+    File results = "${sample_id}.out"
+  }
+  parameter_meta {
+    memory_mb: "Amount of memory to allocate to the JVM"
+    param: "Some arbitrary parameter"
+    sample_id: "The ID of the sample in format foo_bar_baz"
+  }
+  meta {
+    author: "Joe Somebody"
+    email: "joe@company.org"
+  }
+  runtime {
+    container: "my_image:latest"
+    memory: "${memory_mb + 256} MB"
+  }
+}
+```
+
+#### Example 4: BWA MEM
+
+```wdl
+task bwa_mem_tool {
+  input {
+    Int threads
+    Int min_seed_length
+    Array[Int]+ min_std_max_min
+    File reference
+    File reads
+  }
+  command {
+    bwa mem -t ${threads} \
+            -k ${min_seed_length} \
+            -I ${sep=',' min_std_max_min} \
+            ${reference} \
+            ${sep=' ' reads+} > output.sam
+  }
+  output {
+    File sam = "output.sam"
+  }
+  runtime {
+    container: "my_image:latest"
```

Comment:
`myimage:latest` -> `quay.io/biocontainers/bwa:0.7.17--hed695b0_7`
This will make the example fully working.

Reply:
done